plugin.go

```diff
@@ -559,3 +561,133 @@ func (p *ormPlugin) generateListHandler(message *generator.Descriptor) {
+						}
+					} else if tagType[0] == "association_foreignkey" {
+						parentFields = []string{}
+						for _, fName := range strings.Split(tagType[1], ",") { // for compound key
+							parentFields = append(parentFields, lintName(generator.CamelCase(fName)))
+						}
+					}
+				}
+			} // End 'gorm' tag exists within gorm options
+		} // End gorm options exist
+	} // End field options of any kind exist
+	// Resort to default FK: ParentTypeID
+	if len(childFields) != len(parentFields) {
+		// Something's wrong here, mismatched number of primary keys vs. foreign keys
+	} else {
+		for i, child := range childFields {
+			keyMap[child] = parentFields[i]
+		}
+	}
+	return keyMap
+}
+
+func guessZeroValue(typeName string) string {
+	typeName = strings.ToLower(typeName)
+	if strings.Contains(typeName, "string") {
+		return `""`
+	}
+	if strings.Contains(typeName, "int") {
+		return `0`
+	}
+	return ``
+}
+
+func (p *ormPlugin) removeNestedAssociations(message *generator.Descriptor) {
+	typeNamePb := generator.CamelCaseSlice(message.TypeName())
+	typeName := lintName(typeNamePb)
+	if _, exists := typeNames[typeName]; exists {
+		for _, field := range message.Field {
+			if field.IsRepeated() { // Slice, check that it's ORMable, then do stuff
+				fieldType, _ := p.GoType(message, field)
+				rawFieldType := strings.Trim(fieldType, "[]*")
+				if _, exists := convertibleTypes[rawFieldType]; exists {
+
+					keys := findAssociationKeys(message, typeNames[typeName], field)
+					childFKeyTypeName := ""
+					p.P(`filterObj`, rawFieldType, ` := `, rawFieldType, `{}`)
+					for k, v := range keys {
+						for _, childField := range typeNames[rawFieldType].GetField() {
+							if strings.EqualFold(childField.GetName(), k) {
+								childFKeyTypeName, _ = p.GoType(message, childField)
+								break
+							}
+						}
+						if strings.Contains(childFKeyTypeName, "*") { // pointer as key type is strange
+							p.P(`if ormObj.`, v, ` == nil || ormObj.`, v, ` == `, guessZeroValue(childFKeyTypeName), ` {`)
+						} else {
+							p.P(`if ormObj.`, v, ` == `, guessZeroValue(childFKeyTypeName), ` {`)
+						}
+						p.In()
+						p.P(`tx.Rollback()`)
+						p.P(`return nil, fmt.Errorf("Can't do overwriting update with no '`, v,
+							`' value for FK of field '`, lintName(generator.CamelCase(field.GetName())), `'")`)
+						p.P(`}`)
+
+						p.P(`filterObj`, rawFieldType, `.`, k, ` = ormObj.`, v)
+					}
+					p.P(`if err := tx.Where(filterObj`, rawFieldType, `).Delete(`, strings.Trim(fieldType, "[]*"), `{}).Error; err != nil {`)
+					p.P(`tx.Rollback()`)
+					p.P(`return nil, err`)
+					p.P(`}`)
+				}
+			}
+		}
+	}
+}
+
+func (p *ormPlugin) generateUpdateWithOverwriteHandler(message *generator.Descriptor) {
+	typeNamePb := generator.CamelCaseSlice(message.TypeName())
+	typeName := lintName(typeNamePb)
+	p.P(`// DefaultCascadedUpdate`, typeName, ` executes a basic gorm update call`)
```

Comment:
I think this comment needs to be expanded to indicate that it's different from a "basic gorm update call".

I'm wondering if it'd be cleaner to have a single `DefaultUpdateX(ctx context.Context, in *X, db *gorm.db, removeMissingChildren bool)` function instead of two separate functions. Just a simple style suggestion -- I'll leave the ultimate decision to you